openml/datasets/functions.py

```diff
@@ -236,59 +247,75 @@ def get_dataset(did):
         raise ValueError("Dataset ID is neither an Integer nor can be "
                          "cast to an Integer.")
 
-    description = _get_dataset_description(did)
-    arff_file = _get_dataset_arff(did, description=description)
+    did_cache_dir = _create_dataset_cache_directory(did)
+
+    try:
+        description = _get_dataset_description(did_cache_dir, did)
+        arff_file = _get_dataset_arff(did_cache_dir, description)
+        # TODO not used yet, figure out what to do with them...
+        features = _get_dataset_features(did_cache_dir, did)
+        qualities = _get_dataset_qualities(did_cache_dir, did)
+    except Exception as e:
```

Comment:
I'm not sure what kind of exceptions can happen. But for all exceptions, the cache dir might be in an undefined state, thus I want to catch them all.